File: Jaydeep/DNN Training/JRC Modules/bhattji_ProbDriversC/UpdateProblemDrivers.cs
using System;
//using System.Collections.Generic;
using DotNetNuke.Services.Scheduling;
using DotNetNuke.Services.Exceptions;

namespace bhattji.Scheduler.ProblemDrivers
{
    public class UpdateProblemDrivers : SchedulerClient
    {
        public UpdateProblemDrivers(ScheduleHistoryItem objScheduleHistoryItem)
            : base()
        {
            this.ScheduleHistoryItem = objScheduleHistoryItem;
        }

        public override void DoWork()
        {
            string ScheduleNote = "";
            try
            {
                //Perform required items for logging
                this.Progressing();

                //Your code goes here
                //Update SalesPersons For ProbDriver
                bhattji.Modules.SalesPersons.Business.SalesPersonsController objSPC = new bhattji.Modules.SalesPersons.Business.SalesPersonsController();
                objSPC.UpdateSalesPersons4ProbDriver();
                ScheduleNote += "Updated Drivers for 'Problem Driver' Flag. \n";

                //Update InterOffices For HasProbDrivers
                bhattji.Modules.InterOffices.Business.InterOfficesController objIOC = new bhattji.Modules.InterOffices.Business.InterOfficesController();
                objIOC.UpdateInterOffices4HasProbDrivers();
                ScheduleNote += "Updated InterOffices for 'Has Problem Drivers' Flag. \n";

                //Show success
                this.ScheduleHistoryItem.Succeeded = true;

                //To log note
                this.ScheduleHistoryItem.AddLogNote(ScheduleNote);

            }
            catch (Exception ex)
            {
                //Show failure
                this.ScheduleHistoryItem.Succeeded = false;

                // log the exception into the scheduler framework
                ScheduleNote += "EXCEPTION: " + ex.ToString();
                this.ScheduleHistoryItem.AddLogNote(ScheduleNote);
                //InsertLogNote("Exception = " & ex.ToString())

                // call the Errored method
                this.Errored(ref ex);

                // log the exception into the DNN core
                Exceptions.LogException(ex);

            }
        }
    }
}

File: Jaydeep/DNN Training/JRC Modules/bhattji_SalesPersons/Components/UpdateProblemDrivers.vb
'using System.Collections.Generic;
Imports DotNetNuke.Services.Scheduling
Imports DotNetNuke.Services.Exceptions

Namespace bhattji.Scheduler.ProblemDrivers
    Public Class UpdateProblemDrivers
        Inherits SchedulerClient
        Public Sub New(ByVal objScheduleHistoryItem As ScheduleHistoryItem)
            MyBase.New()
            Me.ScheduleHistoryItem = objScheduleHistoryItem
        End Sub

        Public Overrides Sub DoWork()
            Dim ScheduleNote As String = ""
            Try
                'Perform required items for logging
                Me.Progressing()

                'Your code goes here
                'Update SalesPersons For ProbDriver
                Dim objSPC As New bhattji.Modules.SalesPersons.Business.SalesPersonsController()
                objSPC.UpdateSalesPersons4ProbDriver()
                ScheduleNote &= "Updated Drivers for 'Problem Driver' Flag. " & vbCrLf

                'Update InterOffices For HasProbDrivers
                Dim objIOC As New bhattji.Modules.InterOffices.Business.InterOfficesController()
                objIOC.UpdateInterOffices4HasProbDrivers()
                ScheduleNote &= "Updated InterOffices for 'Has Problem Drivers' Flag. " & vbCrLf

                'Show success
                Me.ScheduleHistoryItem.Succeeded = True

                'To log note

                Me.ScheduleHistoryItem.AddLogNote(ScheduleNote)
            Catch ex As Exception
                'Show failure
                Me.ScheduleHistoryItem.Succeeded = False

                ' log the exception into the scheduler framework
                ScheduleNote &= "EXCEPTION: " & ex.ToString()
                Me.ScheduleHistoryItem.AddLogNote(ScheduleNote)
                'InsertLogNote("Exception = " & ex.ToString())

                ' call the Errored method
                Me.Errored(ex)

                ' log the exception into the DNN core

                Exceptions.LogException(ex)
            End Try
        End Sub
    End Class
End Namespace
